src/Akka.Persistence.Sql.Linq2Db/Journal/DAO/BaseByteArrayJournalDao.cs

```diff
@@ -374,31 +374,75 @@ await db.GetTable<JournalMetaData>()
                     SequenceNumber = md.sequenceNumber,
                     PersistenceId = md.persistenceId
                 };
-        private IQueryable<long> MaxSeqNumberForPersistenceIdQuery(
+        private IQueryable<long?> MaxSeqNumberForPersistenceIdQuery(
             DataConnection db, string persistenceId, long minSequenceNumber = 0)
         {
-            
-            
-
-            var queryable = db.GetTable<JournalRow>()
-                .Where(r => r.persistenceId == persistenceId).Select(rowDataSelector);
             if (minSequenceNumber != 0)
             {
-                queryable = queryable.Where(r =>
-                    r.SequenceNumber > minSequenceNumber);
+                if ((_journalConfig.DaoConfig.SqlCommonCompatibilityMode))
```

Comment:
So, this was kinda ugly to make it work everywhere in a way that:

 - Gave reasonably Clean/Sane SQL in all scenarios
 - Had C# code that was reasonably sane to read
 - Wasn't disgusting in IL

Based on testing, having separate methods like such gives us more consistent performance in recovery, and is not that much worse to maintain.

Reply:
I think it's probably fine